src/BlocksRuntime/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+
+cmake_minimum_required(VERSION 3.4.3)
+project(BlocksRuntime
+        LANGUAGES C)
+
+include(CheckIncludeFiles)
+include(GNUInstallDirs)
+
+if(CMAKE_C_SIMULATE_ID STREQUAL MSVC)
+  # clang-cl interprets paths starting with /U as macro undefines, so we need to
+  # put a -- before the input file path to force it to be treated as a path.
+  string(REPLACE "-c <SOURCE>" "-c -- <SOURCE>" CMAKE_C_COMPILE_OBJECT "${CMAKE_C_COMPILE_OBJECT}")
+endif()
+
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_C_STANDARD_REQUIRED YES)
+
+set(CMAKE_C_VISIBILITY_PRESET hidden)
+set(CMAKE_C_VISIBILITY_INLINES_HIDDEN YES)
+
+check_include_files("objc/objc-internal.h" HAVE_OBJC)
+
+option(INSTALL_PRIVATE_HEADERS "install private headers in the same location as the public ones" NO)
```

Comment:
By placing this inside an ExternalProject there's no way to trigger this option anymore.

Reply:
Oh, but there is!  There is the still the option in libdispatch, which just passes it through :-)